blabla/sentence_aggregators/lexico_semantic_fearture_aggregator.py

```diff
@@ -274,6 +514,101 @@ def handle(self):
             return tot_num_prons / tot_num_nouns
         return NOT_AVAILABLE
 
+class Total_Dependency_Distance(object):
+    """Class to calculate the sum of dependency distances
+        Ref: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5337522/
+    """
+
+    def __init__(self, sentence_objs):
+        """The init method to initialize with an array of sentence objects
+        """
+        self.sentence_objs = sentence_objs
+
+    def handle(self):
+        """Method to calculcate the total dependency distance across all sentences
+            Args:
+                None
+            Returns:
+                the sum of dependency distances
+        """
+        tot_dist = 0
+        for so in self.sentence_objs:
+            sd = so.stanza_doc.to_dict()[0]
+            tot_dist += np.sum([abs(int(dep['id']) - dep['head']) for dep in sd])
+        return tot_dist
+
+class Average_Dependency_Distance(object):
+    """Class to calculate the sum of dependency distances
+        Ref: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5337522/
+    """
+
+    def __init__(self, sentence_objs):
+        """The init method to initialize with an array of sentence objects
+        """
+        self.sentence_objs = sentence_objs
+
+    def handle(self):
+        """Method to calculcate the total dependency distance across all sentences
+            Args:
+                None
+            Returns:
+                the sum of dependency distances
+        """
+        tot_dist = []
+        for so in self.sentence_objs:
+            sd = so.stanza_doc.to_dict()[0]
+            tot_dist.append(sum([abs(int(dep['id']) - dep['head']) for dep in sd]))
+        return np.mean(tot_dist)
```

Comment:
@Sarenne , should we add a check to verify that the list is populated before calculating the mean? Otherwise may result in nan.

Reply:
Yes! Good call.